src/combinatorics/simple_graph/connectivity.lean

```diff
@@ -85,6 +85,10 @@ attribute [refl] walk.nil
 
 @[simps] instance walk.inhabited (v : V) : inhabited (G.walk v v) := ⟨walk.nil⟩
 
+/-- The one-edge walk associated to a pair of adjacent vertices. -/
+@[pattern, reducible] def adj.to_walk {G : simple_graph V} {u v : V} (h : G.adj u v) :
```

Comment:
I'm slightly nervous about `reducible` tags - what's the advantage of this one?

Reply:
This is meant to be an abbreviation for `walk.cons _ walk.nil`, for convenience. We don't want to have to write API for this, and I didn't just use `abbreviation` to make sure the equation compiler is used.